src/d3a/models/strategy/market_maker_strategy.py

```diff
@@ -30,9 +30,10 @@ def __init__(self, energy_rate_profile=None, energy_rate=None, grid_connected=Tr
         if energy_rate_profile is not None:
             energy_rate = read_and_convert_identity_profile_to_float(energy_rate_profile)
         else:
-            energy_rate = ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE \
+            energy_rate = read_and_convert_identity_profile_to_float(
+                ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE) \
                 if energy_rate is None else read_and_convert_identity_profile_to_float(energy_rate)
-        GlobalConfig.market_maker_rate = energy_rate
+            GlobalConfig.market_maker_rate = energy_rate
```

Comment:
Why do you only update the Globalconfig if `energy_rate_profile is None` ?

Reply:
Nice catch, was remnant of testing, reverted now to out of if else block